src/ui/public/kuery/ast/kql.peg

```diff
@@ -1,123 +1,147 @@
-/*
- * Kuery parser
- */
-
-/*
- * Initialization block
- */
+// Initialization block
 {
-  var nodeTypes = options.helpers.nodeTypes;
+  var { nodeTypes: { function: { buildNode }}} = options.helpers;
+}
+
+start
+  = OrQuery
+  / EmptyQuery
 
-  if (options.includeMetadata === undefined) {
-    options.includeMetadata = true;
+EmptyQuery
+  = Space* {
+    return buildNode('and', []);
   }
 
-  function addMeta(source, text, location) {
-    if (options.includeMetadata) {
-      return Object.assign(
-        {},
-        source,
-        {
-          text: text,
-          location: simpleLocation(location),
-        }
-      );
-    }
+OrQuery
+  = left:AndQuery Or right:OrQuery {
+    return buildNode('or', [left, right]);
+  }
+  / AndQuery
 
-    return source;
+AndQuery
+  = left:NotQuery And right:AndQuery{
+    return buildNode('and', [left, right]);
   }
+  / NotQuery
 
-  function simpleLocation(location) {
-    // Returns an object representing the position of the function within the expression,
-    // demarcated by the position of its first character and last character. We calculate these values
-    // using the offset because the expression could span multiple lines, and we don't want to deal
-    // with column and line values.
-    return {
-      min: location.start.offset,
-      max: location.end.offset
+NotQuery
+  = Not query:SubQuery {
+    return buildNode('not', query);
+  }
+  / SubQuery
+
+SubQuery
+  = '(' Space* query:OrQuery Space* ')' { return query; }
+  / Expression
+
+Expression
+  = FieldListExpression
+  / FieldRangeExpression
+  / ValueExpression
+
+FieldListExpression
+  = field:Literal Space* ':' Space* list:SubList {
+    function buildNodeWithField(node) {
+      if (node.hasOwnProperty('arguments')) {
+        const args = node.arguments.map(buildNodeWithField);
+        return { ...list, arguments: args };
```

Comment:
I like this a bit more than the old version, though it's still kinda hard to understand. I can't say I have a better idea at the moment though. 

I wonder if at some point it will make sense for the `<field>:(sub-expression)` to have its own node type in the AST, so that the field could be associated with that node rather than the individual `is` nodes. That would make it easier to know how to re-serialize the AST too. Just thinking out loud, it would probably be a waste of time to mess around with that now since this serves our needs fine at the moment.